pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -1467,8 +1488,18 @@ static void Object_beginTypeContext(JSOBJ _obj, JSONTypeContext *tc) {
     tc->type = JT_UTF8;
     return;
   } else if (object_is_decimal_type(obj)) {
-    pc->doubleValue = PyFloat_AsDouble(obj);
-    tc->type = JT_DOUBLE;
+    PyObject *is_nan_py = PyObject_RichCompare(obj, obj, Py_NE);
```

Comment:
Do the test cases cover this code? If not, can you add a NaN case?

Reply:
This test should cover it:

https://github.com/Tolker-KU/pandas/blob/d60f71fe3d20bef8eca0861e742a392dc2c35d4a/pandas/tests/io/json/test_pandas.py#L2027-L2039

https://github.com/Tolker-KU/pandas/blob/d60f71fe3d20bef8eca0861e742a392dc2c35d4a/pandas/conftest.py#L476-L481

https://github.com/Tolker-KU/pandas/blob/d60f71fe3d20bef8eca0861e742a392dc2c35d4a/pandas/_testing/__init__.py#L165